zanata-war/src/main/java/org/zanata/action/ProjectHome.java

```diff
@@ -688,7 +689,7 @@ private void validateProjectState(HProject project) {
             log.warn(
                     "Project [id={}, slug={}], does not exist or is soft deleted: {}",
                     projectId, getSlug(), project);
-            throw new EntityNotFoundException();
+            throw new ProjectNotFoundException();
```

Comment:
What's wrong with EntityNotFoundException for this sort of thing? If we really need to introduce ProjectNotFoundException and VersionNotFoundException, they should probably extend EntityNotFoundException, since they're just a specialisation of the same idea.